integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/K8sEvents.java

```diff
@@ -353,15 +352,35 @@ public static void checkDomainEventContainsExpectedMsg(String opNamespace,
                                                          String type,
                                                          OffsetDateTime timestamp,
                                                          String expectedMsg) {
-    checkEvent(opNamespace, domainNamespace, domainUid, reason, type, timestamp);
-    CoreV1Event event =
-        getEvent(opNamespace, domainNamespace, domainUid, reason, type, timestamp);
-    if (event != null && event.getMessage() != null) {
-      assertTrue(event.getMessage().contains(expectedMsg),
-          String.format("The event message does not contain the expected msg %s", expectedMsg));
-    } else {
-      fail("event is null or event message is null");
-    }
+    withStandardRetryPolicy
+        .conditionEvaluationListener(condition ->
+            getLogger().info("Waiting for domain event {0} to be logged in namespace {1} "
+                    + "(elapsed time {2}ms, remaining time {3}ms)",
+                reason,
+                domainNamespace,
+                condition.getElapsedTimeInMS(),
+                condition.getRemainingTimeInMS()))
+        .until(domainEventContainsExpectedMsg(opNamespace, domainNamespace, domainUid, reason,
+            type, timestamp, expectedMsg));
+  }
+
+  private static Callable<Boolean> domainEventContainsExpectedMsg(String opNamespace,
+                                                                  String domainNamespace,
+                                                                  String domainUid,
+                                                                  String reason,
+                                                                  String type,
+                                                                  OffsetDateTime timestamp,
+                                                                  String expectedMsg) {
+    return () -> {
+      if (domainEventExists(opNamespace, domainNamespace, domainUid, reason, type, timestamp)) {
```

Comment:
The checkDomainEventContainsExpectedMsg() method only checks the first event that matches the event type, reason, timestamp and etc. Multiple DOMAIN_PROCESSING_FAILED type events could be generated during the test time frame. The logic need to make sure that it checks all matching events.

Reply:
fixed